### PR TITLE
Add directive rccif.js and test files for validate cif

### PR DESCRIPTION
**Pull request by JoseAntonioRuiz**
Directive for validate cif number rccif.js  and unit test in karma.

### CODE REVIEW

File: src/directives/rccif.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                if(cif.substr(cif.length-1,1)==String(unit))
+                    return true;
+            }else{
+                //Se valida como un dni
+                return validateDNI(cif);
+            }
+            return false;
+        }
+
+        function validateDNI(dni){
```
Comment (varyvol) [inline src/directives/rccif.js]: I think this function is not used here, so you can remove it.

File: src/directives/rccif.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+			if(fullCIFValidation(currentTypedValue)){
+				ctrl.$setValidity('cif', true);
+			}else{
+				ctrl.$setValidity('cif', false);
+			}
+		};
+
+		/*
+        * Tiene que recibir el cif sin espacios ni guiones
```
Comment (varyvol) [inline src/directives/rccif.js]: Write the comments in english too, as you already did with the code.

File: src/directives/rccif.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        * Esta funcion es llamada
+        * http://www.lawebdelprogramador.com/codigo/JavaScript/1992-Validar_un_CIF_NIF_y_DNI.html
+        */
+
+        function fullCIFValidation(cif){
+        	return (validateCIFSintax(cif) && validateCIFSemantic(cif));
+        }
+        
+        function validateCIFSintax(cif){
+            var cifPattern = /^[ABCDEFGHJKLMNPQRSUVW][0-9]{7}[0-9A-J]$/;
```
Comment (varyvol) [inline src/directives/rccif.js:24]: This sintantic and semantic validations should be in a service ideally, just as we did for the DNI.

File: src/directives/rccif.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+            return cifPattern.test(cif);
+        }
+
+        function validateCIFSemantic(cif)
```
Comment (varyvol) [inline src/directives/rccif.js:29]: This function is really similar to the DNI one, so we should evaluate if we can put it in a common place.

**Comment by brauliodiez** [comment]
Please implement feedback and request a new pull request.

Carlos, should check as well the travis integration, it's failing

File: src/directives/rccif.js (hunk 1 above)
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js]: Ok, I'll change.

File: src/directives/rccif.js (hunk 2 above)
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js]: Ok, I'll change.
Thanks.

File: src/directives/rccif.js (hunk 4 above)
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js:29]: Would be nice but should review the group

File: src/directives/rccif.js (hunk 3 above)
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js:24]: OK, I'll change.

File: src/directives/rccif.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
+directives.directive('rcCif', [function(){
+	return {
+		require:"ngModel",
+	
+	 link: function (scope, elm, attrs, ctrl){
+		var validationToExecute = function (viewValue) {
+			var currentTypedValue = viewValue;
+
```
Comment (anenchu) [inline src/directives/rccif.js]: No need to create a variable for this; just feed viewValue directly to fullCIFValidation

File: src/directives/rccif.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        
+        function validateCIFSintax(cif){
+            var cifPattern = /^[ABCDEFGHJKLMNPQRSUVW][0-9]{7}[0-9A-J]$/;
+
+            return cifPattern.test(cif);
+        }
+
+        function validateCIFSemantic(cif)
+        {
+
```
Comment (anenchu) [inline src/directives/rccif.js]: This function is really long; you could separte calculations in method so we comply with SRP

File: src/directives/rccif.js (hunk 5 above)
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js]: Ok, it's true. I'll remove it.

File: src/directives/rccif.js (hunk 6 above)
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js]: I'll move any calculations in a service.

File: src/directives/rccif.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            var firstChar=cif.substr(0,1).toUpperCase();
+
+            if(firstChar.match(/^[FJKNPQRSUVW]$/))
+            {
+                //Started by .... compared with the last letter
+                if(String.fromCharCode(64+unit).toUpperCase()==cif.substr(cif.length-1,1).toUpperCase())
+                    return true;    
+            }else if(firstChar.match(/^[XYZ]$/)){
+                    var newcif;
```
Comment (JoseAntonioRuiz) [inline src/directives/rccif.js]: This else if isn't necessary because is for calculate the nie.
I´ll delete.

**Comment by JoseAntonioRuiz** [comment]
I have a problem. I try add the service to directive but I can´t and I don´t know why.
I create the service, you can see, but when I add the reference into the directive I appears errors.
Can anyone help me?

**Pull request closed without merging**